pallets/eth-bridge/src/lib.rs

```diff
@@ -544,6 +561,46 @@ pub mod pallet {
                 }
             }
         }
+
+        fn on_idle(_block: BlockNumberFor<T>, mut remaining_weight: Weight) -> Weight {
+            if remaining_weight.checked_reduce(T::DbWeight::get().reads(1)).is_none() {
+                return remaining_weight
+            }
+            remaining_weight.saturating_reduce(T::DbWeight::get().reads(1));
+            let active_range = match Self::active_ethereum_range() {
+                Some(range) => range,
+                None => return remaining_weight,
+            };
+
+            // Ensure enough capacity is there for doing the processing.
+            let partitions_count = match active_range.status {
+                EventProcessingStatus::Accepted(partitions_count) => partitions_count,
+                _ => return remaining_weight,
+            };
+
+            // TODO replace zero with benchmark value
+            let process_weight =
+                T::DbWeight::get().reads(partitions_count as u64).saturating_add(Weight::zero());
+            if remaining_weight.checked_reduce(process_weight).is_none() {
+                return remaining_weight
+            }
+
+            // Read all, then sort, then start processing one by one.
+            let mut partitions =
+                DiscoveredEventsFractions::<T>::iter_key_prefix(&active_range.range)
+                    .collect::<Vec<DiscoveredEthEventsFraction>>();
+            partitions.sort_by(|a, b| a.fraction().cmp(&b.fraction()));
```

Comment:
Is this necessary? Or even desirable? We will be repeating part of the same sorting work every time we run this hook, I believe, if we don't process all of the partitions in the time we have available. Is there a strict need to process the events in any given order?

Reply:
yes, its a requirement to process the events in the right order as some nft operations (transfer of ownership) has different result based on the order.